workers/generic-worker/errorreport/errorreport_test.go

```diff
@@ -24,6 +24,7 @@ func setupProtocols() (*workerproto.Protocol, *workerproto.Protocol) {
 func TestSendEmptyExtra(t *testing.T) {
 	errorReported := false
 	lock := sync.Mutex{}
+	defer lock.Unlock()
```

Comment:
If the routine was to prematurely exit any time after line 27 (`defer lock.Unlock()`), but _before_ the mutex was locked on line 52, we would have a runtime exception due to trying to unlock an unlocked mutex.

There are only two real possible solutions, and they both would be at line 51 where the mutex is locked:

First option:

```
lock.Lock()
defer lock.Unlock()
```

or second option:

```
lock.Lock()
lock.Unlock()
```

I prefer the first of these because it is simpler, and there is no reason for the lock to be held while the test failure condition is checked and reported.

Reply:
We have to do the second to avoid linter errors. We could disable them, but I think this is fine.